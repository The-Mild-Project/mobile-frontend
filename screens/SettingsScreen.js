import React, { useState } from 'react';
import { FlatList, Image, Platform, StyleSheet, View, Text, Button } from 'react-native';
import {ScrollView} from "react-native-gesture-handler";
import useRestaurants from "../hooks/useRestaurants";
import ResultsList from "../components/ResultsList";
import SearchBar from "../components/SearchBar";
import Storage from '../utility/Storage';

const SettingsScreen = ({route, navigation}) => {
    const [term, setTerm] = useState('');
    const name = route.params.name;
    const email = route.params.email;
    const storage = new Storage();

    const logout = () => {
        // remove from storage and redirect to login
        storage.remove("googleId").then(() => {
            navigation.navigate('Welcome')
        })
    }

    return (

        <View style={styles.container}>
            <Text style={styles.name}>Name: {name}</Text>
            <Text style={styles.name}>Email: {email}</Text>
            <Button title="Logout" onPress={() => logout()} />
        </View>
    )
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        paddingTop: 12
    },
    item: {
        padding: 10,
        fontSize: 18,
        height: 44,
    },
    name: {
        fontSize: 20,
        fontWeight: 'bold'
    },
    contentContainer: {
        paddingTop: 30,
    }
});

export default SettingsScreen;