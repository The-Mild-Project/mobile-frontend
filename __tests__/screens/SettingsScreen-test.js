import * as React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import SettingsScreen from '../../screens/SettingsScreen';

jest.mock('expo', () => ({
    AppLoading: 'AppLoading',
}));

describe('App', () => {
    jest.useFakeTimers();

    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it(`renders the loading screen`, () => {
        const tree = renderer.create(<SettingsScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
