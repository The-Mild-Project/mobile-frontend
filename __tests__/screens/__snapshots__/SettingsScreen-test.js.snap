// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the loading screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Name: 
  </Text>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Email: 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</View>
`;
